src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -2898,6 +2898,74 @@ public GHIssueEvent getIssueEvent(long id) throws IOException {
                 .wrapUp(root);
     }
 
+    /**
+     * Lists all the workflows of this repository.
+     *
+     * @param sort
+     *            the sort order. If null, defaults to github api default, currently {@link ForkSort#NEWEST
+     *            ForkSort.NEWEST}.
+     * @return the paged iterable
+     */
+    public PagedIterable<GHWorkflow> listWorkflows() {
+        return root.createRequest()
+                .withUrlPath(getApiTailUrl("actions/workflows"))
+                .toIterable(GHWorkflow[].class, item -> item.wrapUp(root));
+    }
+
+    /**
+     * Gets a workflow by id.
+     *
+     * @param id
+     *            the id of the workflow run
+     * @return the workflow run
+     * @throws IOException
+     *             the io exception
+     */
+    public GHWorkflow getWorkflow(long id) throws IOException {
+        return getWorkflow(String.valueOf(id));
+    }
+
+    /**
+     * Gets a workflow by name of the file.
+     *
+     * @param name
+     *            either the name of the file (e.g. my-workflow.yml) or the id as a string
+     * @return the workflow run
+     * @throws IOException
+     *             the io exception
+     */
+    public GHWorkflow getWorkflow(String nameOrId) throws IOException {
+        return root.createRequest()
+                .withUrlPath(getApiTailUrl("actions/workflows"), nameOrId)
+                .fetch(GHWorkflow.class)
+                .wrapUp(this);
+    }
+
+    /**
+     * Retrieves workflow runs.
+     *
+     * @return the workflow run query builder
+     */
+    public GHWorkflowRunQueryBuilder queryWorkflowRuns() {
+        return new GHWorkflowRunQueryBuilder(this);
+    }
+
+    /**
+     * Gets a workflow run.
+     *
+     * @param id
+     *            the id of the workflow run
+     * @return the workflow run
+     * @throws IOException
+     *             the io exception
+     */
+    public GHWorkflowRun getWorkflowRun(long id) throws IOException {
+        return root.createRequest()
+                .withUrlPath(getApiTailUrl("actions/runs"), String.valueOf(id))
+                .fetch(GHWorkflowRun.class)
+                .wrapUp(this);
+    }
+
```

Comment:
What would you think of having a `workflows()` method that returns a `WorkflowsEndpoint` instance linked to this repository and move all these methods into that class? The idea here is to reduce the size/complexity of `GHRepository` and to more closely match the GraphQL object model. 

I'm not saying you have to do it for this change, I'm more taking notes for v2 design ideas and I'd like your opinion.

Reply:
I think for now it makes sense to have them here given the others are here. But I agree it would be nice to group them into subapis.

Reply:
I could do it now if you would like to do it for new API endpoints, it's not biggie.

Reply:
Meh, let's not innovate right now.

Reply:
OK :).